diff --git a/init/init.c b/init/init.c
index 9ee59bd..3d4f625 100755
--- a/init/init.c
+++ b/init/init.c
@@ -64,6 +64,7 @@ static char bootmode[32];
 static char baseband[32];
 static char carrier[32];
 static char bootloader[32];
+static char delayms[8];
 static char hardware[32];
 static char modelno[32];
 static unsigned revision = 0;
@@ -452,6 +453,8 @@ static void import_kernel_nv(char *name, int in_qemu)
             strlcpy(hardware, value, sizeof(hardware));
         } else if (!strcmp(name,"androidboot.modelno")) {
             strlcpy(modelno, value, sizeof(modelno));
+        } else if (!strcmp(name, "androidboot.delayms")) {
+            strlcpy(delayms, value, sizeof(delayms));
         }
     } else {
         /* in the emulator, export any kernel option with the
@@ -723,6 +726,13 @@ int main(int argc, char **argv)
 
     /* pull the kernel commandline and ramdisk properties file in */
     import_kernel_cmdline(0, import_kernel_nv);
+    {
+        int ms = atoi(delayms);
+        if (ms != 0) {
+            INFO("wait %d ms to continue\n", ms);
+            usleep(ms*1000);
+        }
+    }
     /* don't expose the raw commandline to nonpriv processes */
     chmod("/proc/cmdline", 0440);
     get_hardware_name(hardware, &revision);
